practice-two/src/components/FormValidation/FormValidation.jsx

```diff
@@ -121,12 +111,25 @@ const FormValidation = ({
       setErrors(newErrors);
 
       if (Object.keys(newErrors).length === 0) {
-        const response = await createCustomerService(formData);
         let toastMessage = MESSAGES.GET.SUCCESSES.ADD_SUCCESSED;
-        if (response.error) {
-          toastMessage = MESSAGES.GET.ERRORS.ADD_FAILED;
+        if (selectedCustomer) {
+          // Update customer
+          const response = await updateCustomerService(selectedCustomer.id, formData);
+          if (response.error) {
+            toastMessage = MESSAGES.GET.ERRORS.UPDATE_FAILED;
+          } else {
+            toastMessage = MESSAGES.GET.SUCCESSES.UPDATE_SUCCESSED
+            dispatch(actionReducerCustomer(ACTION_TYPES.UPDATE, response.data));
+            setSelectedCustomer(response.data)
+          }
         } else {
-          dispatch(actionReducerCustomer(ACTION_TYPES.CREATE, response.data));
+          // Create customer
+          const response = await createCustomerService(formData);
```

Comment:
when the customer id create?

Reply:
When creating customer successfully, it will store in mockapit, mockapit will auto create the id for new customer.
When data return from mockapi to local, it will update the ID for new customer